src/equality.rs

```diff
@@ -103,6 +104,78 @@ impl<L: SynthLanguage> Equality<L> {
     pub fn score(&self) -> impl Ord + Debug {
         L::score(&self.lhs, &self.rhs)
     }
+
+    fn mk_runner(
+        &self,
+        egraph: EGraph<L, SynthAnalysis>,
+        limits: Limits,
+    ) -> Runner<L, SynthAnalysis> {
+        Runner::default()
+            .with_scheduler(egg::SimpleScheduler)
+            .with_iter_limit(limits.iter)
+            .with_node_limit(limits.node)
+            .with_time_limit(Duration::from_secs(600))
+            .with_egraph(egraph)
+            .with_expr(&L::instantiate(&self.lhs))
+            .with_expr(&L::instantiate(&self.rhs))
+            .with_hook(|r| {
+                if r.egraph.find(r.roots[0]) == r.egraph.find(r.roots[1]) {
+                    Err("Done".to_owned())
+                } else {
+                    Ok(())
+                }
+            })
+    }
+
+    pub fn is_derivable(&self, ruleset: &Ruleset<L>, limits: Limits) -> bool {
```

Comment:
is this file where this method should belong? why?

Reply:
hmm, yeah I thought about this, I could go either way I think. Is it more a property of a rule whether a given ruleset can derive it or a property of a ruleset that it can derive a given rule?

Reply:
yeah, that's a great point! I am kinda leaning more towards the latter because I feel like derivability is a quality of a ruleset not a rule? But then, the function should not be called `is_derivable`...

Reply:
Ok I'm fine with that! I'll move it and call it `can_derive`

Reply:
`can_derive` is a good name!